apps/users/views.py

```diff
@@ -1,3 +1,152 @@
-from django.shortcuts import render
+from rest_framework import mixins, status
+from rest_framework.parsers import MultiPartParser
+from rest_framework.serializers import Serializer
+from rest_framework.decorators import action
+from rest_framework.exceptions import ValidationError
+from rest_framework.permissions import IsAuthenticated, AllowAny
+from rest_framework.response import Response
+from rest_framework.viewsets import GenericViewSet, ModelViewSet
+
+from django.shortcuts import get_object_or_404
+
+from apps.users.models import User, UserVerification, UserAddress
+from apps.users.serializers import UserSerializer, UserVerificationSerializer, UserEmailSerializer, \
+    UserLoginResetSerializer, UserProfileSerializer, UserPasswordUpdateSerializer, UserAddressSerializer
+
+from apps.common.permisions import IsAdminOrOwner, IsAdminOrItself, ReadOnly
+
 
 # Create your views here.
+
+
+class UserSignupViewSet(GenericViewSet):
+    """
+    Used for creating and verifying new accounts.
+    """
+    serializer_class = UserSerializer
+    queryset = User.objects.all()
+    permission_classes = (AllowAny,)
+
+    @action(detail=False, methods=['POST'])
+    def new(self, request, *args, **kwargs):
+        serializer = self.get_serializer(data=request.data)
+
+        serializer.is_valid(raise_exception=True)
+        validated_data = serializer.validated_data
+
+        raw_password = validated_data.pop('password')
+
+        user, created = User.objects.get_or_create(**validated_data)
+
+        user.set_password(raw_password)
+        user.save()
+
+        data = self.get_serializer(user).data
+        return Response(data, status.HTTP_201_CREATED)
+
+    @action(detail=False, methods=['POST'], url_path='send-verification', serializer_class=UserEmailSerializer)
+    def send_verification(self, request, *args, **kwargs):
```

Comment:
La aceasta etapa userul deja trebuie sa fie creat ca obiect in baza, acest endpoint trebuie sa primeasca id-ul userului prin detail=True